torch/lib/c10d/reducer.hpp

```diff
@@ -25,6 +25,28 @@ constexpr int kDefaultBucketBytesCap = int(25 * 1024 * 1024);
 // Collect runtime stats once for every kDDPRuntimeLoggingSampleRate iterations.
 constexpr int kDDPRuntimeLoggingSampleRate = 100;
 
+// Locates a specific variable by replica index and variable index.
+struct VariableIndex {
+  size_t replica_index;
+  size_t variable_index;
+
+  VariableIndex() = default;
+
+  VariableIndex(size_t replica_index_, size_t variable_index_) {
+    replica_index = replica_index_;
+    variable_index = variable_index_;
+  }
+
+  static size_t hash(const VariableIndex& key) {
```

Comment:
We can probably remove this if we make the data structures to be indexed only by replica_index